internal/blocksync/verify.go

```diff
@@ -0,0 +1,108 @@
+package blocksync
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/tendermint/tendermint/light"
+	"github.com/tendermint/tendermint/types"
+)
+
+func VerifyAdjacent(
```

Comment:
can the light client verifier be reused here instead of duplicating?

Reply:
We cannot simply call the light client `VerifyAdjacent` here as the light client function has time constraints. It verifies whether a block is within the trusted period which may not be true if we are blocksync-ing. That being said, we can always alter the `VerifyAdjacent` function in the light client to conditionally check whether we are are calling it from blocksync or the light client. For now, I added it separately until we converge to the desired verification function.